googleapis/python-aiplatform
# feat(sdk): enable loading both JSON and YAML pipelines IR

Fixes #1088

**Main changes**
This PR switches `load_json` to `load_yaml`. This change is backward-compatible, as YAML is a superset of JSON.

**Other changes**
* Added tests for `load_yaml` to demonstrate that is reads both JSON and YAML.
* Added `pyyaml` as a library dependency.
* Updated copyright where applicable.
* Formatted `tests/unit/aiplatform/test_utils.py` with `black`.

**Manually tested the four primary codepaths**
1. Read JSON locally.
2. Read YAML locally.
3. Read JSON from Google Cloud Storage.
4. Read YAML from Google Cloud Storage.

---

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea

cc: @chensun @ji-yaqi
